File: Algorithms/283. Move Zeroes/javascript/solution3.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    let lastNonZeroFoundAt = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            const temp = nums[lastNonZeroFoundAt];
            nums[lastNonZeroFoundAt] = nums[i];
            nums[i] = temp;
            lastNonZeroFoundAt++;
        }
    }
};

File: Algorithms/283. Move Zeroes/golang/solution3.go
func moveZeroes(nums []int) {
	var lastNonZeroFoundAt int = 0
	for i, num := range nums {
		if num != 0 {
			nums[lastNonZeroFoundAt], nums[i] = num, nums[lastNonZeroFoundAt]
			lastNonZeroFoundAt++
		}
	}
}